src/components/Inputs/ColoredLabel.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import Label from '../Label'
+import ColorPicker from './ColorPicker'
+import Input from './Input'
+
+type ColoredLabelValue = {
+  title: string
+  color: string
+}
+
+interface ColoredLabelProps {
+  value: ColoredLabelValue
+  className?: string
+  onChange: ({ title, color }: ColoredLabelValue) => void
+  disabled?: boolean
+  placeholder?: string
+  id?: string
+}
+
+let ColoredLabel = ({ placeholder, disabled, onChange, value, className, id }: ColoredLabelProps) => {
```

Comment:
> I'd rename that to ColoredLabelInput, as I had to read the code to understand that it's not just the label we're talking about (+ I like to have descriptive names when searching for files in VSCode :)).

👍 

> We're not coherent in our use of Styled Components. Here you declare a variable that is wrapped later on. In the other approach, we use constants only (in this case, we would have a InputContainer carrying the styles.

The first reason why I prefer my approach is, as you said, one less "wrapper" element.

Second reason is that the styles of my React component will always be able to be extended by another component. What I mean:

```tsx
// ComponentName.tsx
let ComponentName = ({ className }: { className?: string }) => {
  return <div className={className}>hello</div>
}

ComponentName = styled(ComponentName)`
  color: red;
`

export default ComponentName
```

```tsx
// Page.tsx
import ComponentName from './ComponentName'

const Page = () => {
  return <ComponentNameStyled />
}

const ComponentNameStyled = styled(ComponentName)`
  color: blue; // The color of the "hello" word will now be blue.
`
```

The reason this is possible is that my approach forces me to pass and attach `className` attribute. I was basically inspired by their official docs regarding the [TypeScript caveat](https://styled-components.com/docs/api#caveat-with-classname).

With the first approach, if we forget to pass the `className`, our component won't be re-styled. I had this problem a few times and it took me a while to figure out what is going on:

```tsx
// ComponentName.tsx
const ComponentName = () => {
  return <ComponentNameWrapper>hello<ComponentNameWrapper>
}

const ComponentNameWrapper = styled.div`
  color: red;
`

export default ComponentName
```

```tsx
// Page.tsx
import ComponentName from './ComponentName'

const Page = () => {
  return <ComponentNameStyled />
}

const ComponentNameStyled = styled(ComponentName)`
  color: blue;  // Color will still be red, since `className` was not passed.
`
```

Reply:
Yes, it's an interesting point. For now, I selectively added the `className` prop when needed.
If we decide to go with "your" strategy, we should.. well, adapt all our components for consistency sake ⭐

Reply:
I agree, it makes sense to follow one strategy. What are the arguments against "my" approach? Not sure I understand your argument of _"the other one is more immutable and systematic"_.

Also, whichever approach we decide upon, we don't need to change the whole code base in one go. We can change it whenever we face a component that violates our decision while working on new features. Otherwise it'll be a very boring task 😅

Reply:
Let's do that then! When we come across new components or components that need to be updated, we switch to the "wrapping" strategy.

Reply:
As in "your" approach (option 1)